spark3/src/main/java/org/apache/iceberg/spark/CreateActions.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.connector.catalog.Identifier;
+
+public class CreateActions {
```

Comment:
I'm finding it hard to follow when this would migrate a table vs snapshot a table, and it also requires a separate class.

Did you consider an API using verbs directly? I'm imagining something like this:

```java
Actions.migrate("db.table").execute();
Actions.snapshot("db.table").as("db.table_iceberg").execute()
```

We could add methods as well, but those verbs correspond to the SQL that we expose and we haven't needed anything more complicated than that.

Reply:
If we wanted the most limited api I think we want
Migrate - Takes one arg (identifier)
Snapshot - Takes two args (identifier source, identifier dest, location)

I'm a little worried about having a "snapshot" verb with  "as" since I feel like it also requires an "at" for a location and then I have to do a runtime check to make sure they are both used unless I make a series of chaining classes.

SnapshotStart ---  .as(identifier) -->  SnapshotWithDest --- at(location) --> CreateAction

Although I'm ok with that too, that's probably more complicated for us, but more difficult for an end user to use incorrectly or get an unexpected result from.

This would be the minimal setup then, I'm not sure what createTable like would do?

Reply:
I took a more locked down approach to this for the mean time. We only expose a
Actions.migrate
Actions.snapshot

To the public, this class instead becomes package protected with a few public methods, 
Those inherited from Action
withProperty/ies : for adding table properties
as : for changing the destination